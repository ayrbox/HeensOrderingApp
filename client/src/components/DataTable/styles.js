const styles = {
  actionCell: {
    width: '200px',
  },
};

export default styles;
