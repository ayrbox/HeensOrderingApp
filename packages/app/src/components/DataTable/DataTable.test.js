import React from 'react';
import {
  render,
} from '@testing-library/react';
import DataTable from './DataTable';

describe('<DataTable />', () => {
  let container;
  const mockView = jest.fn();
  const mockEdit = jest.fn();
  const mockDelete = jest.fn();
  const data = [{
    id: '2093482038',
    name: 'Test1',
    description: 'Test1 Description',
  },
  {
    id: '23947829374',
    name: 'Test2',
    description: 'Test2 Description',
  }];
  const columns = [{
    name: 'id',
    label: 'Id column',
    key: true,
    hidden: true,
  }, {
    name: 'name',
    label: 'Name',
  }, {
    name: 'description',
    label: 'Description',
  }];

  beforeEach(() => {
    container = render(
      <DataTable
        data={data}
        columns={columns}
        onView={mockView}
        onEdit={mockEdit}
        onDelete={mockDelete}
      />,
    );
  });

  describe('Render', () => {
    it('renders correctly', () => {
      expect(container).toMatchSnapshot();
    });

    it('DEBUG', () => {
      const { debug } = container;
      debug();
    });

    it.todo('renders headers correctly');
    it.todo('renders rows number correctly');
  });
  
  describe('Behaviour', () => {
    describe('onClick of view button', () => {
      it.todo('should call onView action');
    });

    describe('onClick of edit button', () => {
      it.todo('should call onEdit action');
    });

    describe('onClick of delete button', () => {
      it.todo('should call onDelete action');
    });
  });
});
