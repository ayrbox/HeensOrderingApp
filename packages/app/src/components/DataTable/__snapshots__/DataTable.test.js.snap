// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataTable /> Render renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="MuiTable-root-2"
      >
        <thead
          class="MuiTableHead-root-3"
        >
          <tr
            class="MuiTableRow-root-4 MuiTableRow-head-7"
          >
            <th
              class="MuiTableCell-root-9 MuiTableCell-head-10"
              scope="col"
            >
              Name
            </th>
            <th
              class="MuiTableCell-root-9 MuiTableCell-head-10"
              scope="col"
            >
              Description
            </th>
            <th
              class="MuiTableCell-root-9 MuiTableCell-head-10"
              scope="col"
            />
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root-21"
        >
          <tr
            class="MuiTableRow-root-4"
          >
            <td
              class="MuiTableCell-root-9 MuiTableCell-body-11"
            >
              Test1
            </td>
            <td
              class="MuiTableCell-root-9 MuiTableCell-body-11"
            >
              Test1 Description
            </td>
            <td
              class="MuiTableCell-root-9 MuiTableCell-body-11 DataTable-actionCell-1"
            >
              <button
                aria-label="View"
                class="MuiButtonBase-root-28 MuiIconButton-root-22"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label-27"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-31"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                    <path
                      d="M12 10h-2v2H9v-2H7V9h2V7h1v2h2v1z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root-40"
                />
              </button>
              <button
                aria-label="Edit"
                class="MuiButtonBase-root-28 MuiIconButton-root-22"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label-27"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-31"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root-40"
                />
              </button>
              <button
                aria-label="Delete"
                class="MuiButtonBase-root-28 MuiIconButton-root-22"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label-27"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-31"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root-40"
                />
              </button>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root-4"
          >
            <td
              class="MuiTableCell-root-9 MuiTableCell-body-11"
            >
              Test2
            </td>
            <td
              class="MuiTableCell-root-9 MuiTableCell-body-11"
            >
              Test2 Description
            </td>
            <td
              class="MuiTableCell-root-9 MuiTableCell-body-11 DataTable-actionCell-1"
            >
              <button
                aria-label="View"
                class="MuiButtonBase-root-28 MuiIconButton-root-22"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label-27"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-31"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                    <path
                      d="M12 10h-2v2H9v-2H7V9h2V7h1v2h2v1z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root-40"
                />
              </button>
              <button
                aria-label="Edit"
                class="MuiButtonBase-root-28 MuiIconButton-root-22"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label-27"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-31"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root-40"
                />
              </button>
              <button
                aria-label="Delete"
                class="MuiButtonBase-root-28 MuiIconButton-root-22"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label-27"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-31"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root-40"
                />
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="MuiTable-root-2"
    >
      <thead
        class="MuiTableHead-root-3"
      >
        <tr
          class="MuiTableRow-root-4 MuiTableRow-head-7"
        >
          <th
            class="MuiTableCell-root-9 MuiTableCell-head-10"
            scope="col"
          >
            Name
          </th>
          <th
            class="MuiTableCell-root-9 MuiTableCell-head-10"
            scope="col"
          >
            Description
          </th>
          <th
            class="MuiTableCell-root-9 MuiTableCell-head-10"
            scope="col"
          />
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root-21"
      >
        <tr
          class="MuiTableRow-root-4"
        >
          <td
            class="MuiTableCell-root-9 MuiTableCell-body-11"
          >
            Test1
          </td>
          <td
            class="MuiTableCell-root-9 MuiTableCell-body-11"
          >
            Test1 Description
          </td>
          <td
            class="MuiTableCell-root-9 MuiTableCell-body-11 DataTable-actionCell-1"
          >
            <button
              aria-label="View"
              class="MuiButtonBase-root-28 MuiIconButton-root-22"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label-27"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-31"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M12 10h-2v2H9v-2H7V9h2V7h1v2h2v1z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root-40"
              />
            </button>
            <button
              aria-label="Edit"
              class="MuiButtonBase-root-28 MuiIconButton-root-22"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label-27"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-31"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root-40"
              />
            </button>
            <button
              aria-label="Delete"
              class="MuiButtonBase-root-28 MuiIconButton-root-22"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label-27"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-31"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root-40"
              />
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root-4"
        >
          <td
            class="MuiTableCell-root-9 MuiTableCell-body-11"
          >
            Test2
          </td>
          <td
            class="MuiTableCell-root-9 MuiTableCell-body-11"
          >
            Test2 Description
          </td>
          <td
            class="MuiTableCell-root-9 MuiTableCell-body-11 DataTable-actionCell-1"
          >
            <button
              aria-label="View"
              class="MuiButtonBase-root-28 MuiIconButton-root-22"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label-27"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-31"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M12 10h-2v2H9v-2H7V9h2V7h1v2h2v1z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root-40"
              />
            </button>
            <button
              aria-label="Edit"
              class="MuiButtonBase-root-28 MuiIconButton-root-22"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label-27"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-31"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root-40"
              />
            </button>
            <button
              aria-label="Delete"
              class="MuiButtonBase-root-28 MuiIconButton-root-22"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label-27"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-31"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root-40"
              />
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<DataTable /> renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="MuiTable-root-2"
      >
        <thead
          class="MuiTableHead-root-3"
        >
          <tr
            class="MuiTableRow-root-4 MuiTableRow-head-7"
          >
            <th
              class="MuiTableCell-root-9 MuiTableCell-head-10"
              scope="col"
            >
              Name
            </th>
            <th
              class="MuiTableCell-root-9 MuiTableCell-head-10"
              scope="col"
            >
              Description
            </th>
            <th
              class="MuiTableCell-root-9 MuiTableCell-head-10"
              scope="col"
            />
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root-21"
        >
          <tr
            class="MuiTableRow-root-4"
          >
            <td
              class="MuiTableCell-root-9 MuiTableCell-body-11"
            >
              Test1
            </td>
            <td
              class="MuiTableCell-root-9 MuiTableCell-body-11"
            >
              Test1 Description
            </td>
            <td
              class="MuiTableCell-root-9 MuiTableCell-body-11 DataTable-actionCell-1"
            >
              <button
                aria-label="View"
                class="MuiButtonBase-root-28 MuiIconButton-root-22"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label-27"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-31"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                    <path
                      d="M12 10h-2v2H9v-2H7V9h2V7h1v2h2v1z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root-40"
                />
              </button>
              <button
                aria-label="Edit"
                class="MuiButtonBase-root-28 MuiIconButton-root-22"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label-27"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-31"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root-40"
                />
              </button>
              <button
                aria-label="Delete"
                class="MuiButtonBase-root-28 MuiIconButton-root-22"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label-27"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-31"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root-40"
                />
              </button>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root-4"
          >
            <td
              class="MuiTableCell-root-9 MuiTableCell-body-11"
            >
              Test2
            </td>
            <td
              class="MuiTableCell-root-9 MuiTableCell-body-11"
            >
              Test2 Description
            </td>
            <td
              class="MuiTableCell-root-9 MuiTableCell-body-11 DataTable-actionCell-1"
            >
              <button
                aria-label="View"
                class="MuiButtonBase-root-28 MuiIconButton-root-22"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label-27"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-31"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                    <path
                      d="M12 10h-2v2H9v-2H7V9h2V7h1v2h2v1z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root-40"
                />
              </button>
              <button
                aria-label="Edit"
                class="MuiButtonBase-root-28 MuiIconButton-root-22"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label-27"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-31"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root-40"
                />
              </button>
              <button
                aria-label="Delete"
                class="MuiButtonBase-root-28 MuiIconButton-root-22"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label-27"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-31"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root-40"
                />
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="MuiTable-root-2"
    >
      <thead
        class="MuiTableHead-root-3"
      >
        <tr
          class="MuiTableRow-root-4 MuiTableRow-head-7"
        >
          <th
            class="MuiTableCell-root-9 MuiTableCell-head-10"
            scope="col"
          >
            Name
          </th>
          <th
            class="MuiTableCell-root-9 MuiTableCell-head-10"
            scope="col"
          >
            Description
          </th>
          <th
            class="MuiTableCell-root-9 MuiTableCell-head-10"
            scope="col"
          />
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root-21"
      >
        <tr
          class="MuiTableRow-root-4"
        >
          <td
            class="MuiTableCell-root-9 MuiTableCell-body-11"
          >
            Test1
          </td>
          <td
            class="MuiTableCell-root-9 MuiTableCell-body-11"
          >
            Test1 Description
          </td>
          <td
            class="MuiTableCell-root-9 MuiTableCell-body-11 DataTable-actionCell-1"
          >
            <button
              aria-label="View"
              class="MuiButtonBase-root-28 MuiIconButton-root-22"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label-27"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-31"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M12 10h-2v2H9v-2H7V9h2V7h1v2h2v1z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root-40"
              />
            </button>
            <button
              aria-label="Edit"
              class="MuiButtonBase-root-28 MuiIconButton-root-22"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label-27"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-31"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root-40"
              />
            </button>
            <button
              aria-label="Delete"
              class="MuiButtonBase-root-28 MuiIconButton-root-22"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label-27"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-31"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root-40"
              />
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root-4"
        >
          <td
            class="MuiTableCell-root-9 MuiTableCell-body-11"
          >
            Test2
          </td>
          <td
            class="MuiTableCell-root-9 MuiTableCell-body-11"
          >
            Test2 Description
          </td>
          <td
            class="MuiTableCell-root-9 MuiTableCell-body-11 DataTable-actionCell-1"
          >
            <button
              aria-label="View"
              class="MuiButtonBase-root-28 MuiIconButton-root-22"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label-27"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-31"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M12 10h-2v2H9v-2H7V9h2V7h1v2h2v1z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root-40"
              />
            </button>
            <button
              aria-label="Edit"
              class="MuiButtonBase-root-28 MuiIconButton-root-22"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label-27"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-31"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root-40"
              />
            </button>
            <button
              aria-label="Delete"
              class="MuiButtonBase-root-28 MuiIconButton-root-22"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label-27"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-31"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root-40"
              />
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
